README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
```

Comment:
If we want to be strict about RFC3986 rules for URLs, we might need to drop the `+` from the workspace address.

I read the [RFC section on hostnames](https://tools.ietf.org/html/rfc3986#section-3.2.2) to say `+` is allowed because it's a "sub-delim" in the "reg-name", but we should see if existing URL parser libraries will choke on it...

Besides the `+`, the [Earthstar workspace address specification](https://github.com/earthstar-project/earthstar/blob/master/docs/specification.md#workspace-addresses) guarantees that it's safe to use in the location part of a URL.

Reply:
Just tried: Javascript's built-in `URL` parser can handle the `+`.

(For that parser you have to trick it by string-replacing the `earthstar://` scheme to `http://` or it acts badly, it must have an allowlist of schemes)

Reply:
Nice seems like the + can stay! Regarding the scheme allowlist, that's built into the browser and in places like Agregore, adding earthstar as a protocol handler will automatically add it to the allow list.